commit 6cdca2c3d655ef19d022fb3d8bcbf63491b79db2
Author: Joy Latten <jmlatten@linux.vnet.ibm.com>
Date:   Wed Mar 19 12:50:14 2014 -0500

    Add flag EVP_MD_FLAG_PKEY_METHOD_SIGNATURE to EVP_MD so that signing method
    comes from key type.
    
    Signed-off-by: Joy Latten <jmlatten@linux.vnet.ibm.com>

diff --git a/e_ibmca.c b/e_ibmca.c
index f3fad35..9353470 100644
--- a/e_ibmca.c
+++ b/e_ibmca.c
@@ -711,7 +711,7 @@ static const EVP_MD ibmca_sha1 = {
 	NID_sha1,
 	NID_sha1WithRSAEncryption,
 	SHA_HASH_LENGTH,
-	0,
+	EVP_MD_FLAG_PKEY_METHOD_SIGNATURE,
 	ibmca_sha1_init,
 	ibmca_sha1_update,
 	ibmca_sha1_final,
@@ -728,7 +728,7 @@ static const EVP_MD ibmca_sha256 = {
 	NID_sha256,
 	NID_sha256WithRSAEncryption,
 	SHA256_HASH_LENGTH,
-	0,
+	EVP_MD_FLAG_PKEY_METHOD_SIGNATURE,
 	ibmca_sha256_init,
 	ibmca_sha256_update,
 	ibmca_sha256_final,
